src/Api/Endpoints/Magic.cs

```diff
@@ -13,12 +15,20 @@ public static class MagicEndpoints
     public const string RateLimiterPolicy = nameof(MagicEndpoints);
 
     public static void AddMagicRateLimiterPolicy(this RateLimiterOptions builder) =>
-        builder.AddFixedWindowLimiter(RateLimiterPolicy, limiter =>
+        builder.AddPolicy(RateLimiterPolicy, context =>
         {
-            limiter.Window = TimeSpan.FromMinutes(5);
-            limiter.PermitLimit = 10;
-            limiter.QueueLimit = 0;
-            limiter.AutoReplenishment = true;
+            using var scope = context.RequestServices.CreateScope();
```

Comment:
You could potentially make it a tad quicker accessing the claim directly:

`CustomClaimTypes.AccountName`

Reply:
Done